af/fastpath/fastpath/db.py

```diff
@@ -156,3 +168,158 @@ def upsert_summary(
         # notification_json = ujson.dumps(notification)
         # q = f"SELECT pg_notify('fastpath', '{notification_json}');"
         # cur.execute(q)
+
+
+# # Clickhouse
+
+
+def query_click(query, query_params):
+    q = click_client.execute(query, query_params, with_column_types=True)
+    rows, coldata = q
+    colnames, coltypes = tuple(zip(*coldata))
+
+    for row in rows:
+        yield dict(zip(colnames, row))
+
+
+def _click_create_table_fastpath():
+    # TODO: table creation should be done before starting workers
+    sql = """
+    CREATE TABLE IF NOT EXISTS fastpath
+    (
+        `measurement_uid` String,
+        `report_id` String,
+        `input` String,
+        `probe_cc` String,
+        `probe_asn` Int32,
+        `test_name` String,
+        `test_start_time` DateTime,
+        `measurement_start_time` DateTime,
+        `filename` String,
+        `scores` String,
+        `platform` String,
+        `anomaly` String,
+        `confirmed` String,
+        `msm_failure` String,
+        `domain` String,
+        `software_name` String,
+        `software_version` String,
+        `control_failure` String,
+        `blocking_general` Float32,
+        `is_ssl_expected` Int8,
+        `page_len` Int32,
+        `page_len_ratio` Float32,
+        `server_cc` String,
+        `server_asn` Int8,
+        `server_as_name` String
+    )
```

Comment:
A more general consideration I have is that currently the additional fields which have been added are very web_connectivity specific and tests that are not web_connectivity will have these fields unset. I can imagine that in the future we are going to need to also implement some feature extraction that is specific for particular test types (ex. number of reachable directory authorities in the `tor` test) and I wonder how this is going to impact the evolution of the schema.
I can see one approach, which is the direction where we are going now, of adding everything to the fastpath table, but we do run the risk of the schema eventually becoming a bit bloated. Another approach could potentially be that of putting the test specific fields in dedicated tables and looking them up only when necessary via cross table joins.